### ISSUE TITLE
Sort by multiple columns

### ISSUE THREAD

**Issue by marctich** [trailblazer/trailblazer-finder#7]
Currently sorting only works for a single column, how old school.

**Comment by marctich** [comment]
- Added sorting by multiple columns/directions for Hash, ActiveRecord and Sequel. Temporarily disabled sorting for DataMapper Adapter until i can find the time this weekend to update that Adapter as well.
- Due to the above, a big change in the way sorting works, and which methods are available for it

### Sorting
Really simple sorting feature, fixing the pain of dealing with sorting attributes and directions. Can sort by multiple columns/directions.

NOTE: AS OF 0.1.3 TEMPORARY NOT WORKING FOR DATAMAPPER ADAPTER, WE ARE WORKING ON FIXING IT

#### Sorting Example
```ruby
class Post::Finder < Trailblazer::Finder
  features Sorting

  sortable_by :name, :body
end
```

This feature extends the result[:finder] object with the following methods
```ruby
.results                                # => Posts sorted by title DESC

# Smart sort checking
.sort?('title')                         # => true

# Helpers for seeing current sort direction
.sort_direction_for('title')            # => 'asc'
.sort_direction_for('body')           # => 'desc'

# Helpers for seeing reversing sort direction
.reverse_sort_direction_for('title')            # => 'desc'
.reverse_sort_direction_for('body')          # => 'asc'

# Params for sorting links (new if none exists, existing params if exists)
.sort_params_for('title')

# Add Params for sorting links (add to existing / replace with different direction)
.add_sort_params_for('title')

# New Params for sorting links (reset)
.new_sort_params_for('title')
```

Basically there is a default filter called sort where one can now put all (before it was a single column only) the sort columns + directions as follows:
`sort=id asc, name desc, price asc`

**Comment by marctich** [comment]
For some reason 2.1 build failed and returned a totally different result in sorting then on other versions, since ruby 2.1 support is officially ended anyway, i upped ruby's version requirement to 2.2. 

If at some point someone shows up who does us 2.1 and wants to use it, I'll look into it, until then I can't be bothered.

**Comment by marctich** [comment]
https://github.com/trailblazer/trailblazer-finder/issues/8

https://github.com/trailblazer/trailblazer-finder/commit/c2fa669c47d90cd1086ac5e93983809a471983ba

Only outstanding issue now is the 2.1 oddball default sort order for array/hash.
